### PR TITLE
Electron preference management for font selection

### PR DESCRIPTION
**Pull request by timtmok**
### Intent
Addresses #10438 so that it can read the existing preferences that Qt writes. This allows checking if there is an existing font selected. The editor will now show the selected font and the appearance options will have the font pre-selected.

Also fixed an issue with Linux loading the font list. `node-system-fonts` has a bug with how it uses Fontconfig to query all available fonts. It sometimes incorrectly marks a font as monospace. The bug is not present when providing query options.

### Approach
For MacOS, it uses Electron's `systemPreferences`. The `CFBundleIdentifier` needed to be properly set for this to work. In local dev, this is not set so it reads from `com.github.Electron`.

Windows and Linux both read from `desktop.ini`. A package has been added that can parse ini files. Both platforms use the same code but use the platform specific location for `desktop.ini`.

There is an interface for the preference manager. It uses Electron's preference value types. This might be able to be cleaned up a bit to use something more agnostic to all platforms. For Linux/Windows, it will always be a `string`.

### Automated Tests
WIP

### QA Notes
 Linux should only show monospace fonts in the options. All platforms should load the font specified in the preference store. Setting a font is still unsupported.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

File: src/node/desktop/src/main/preferences/mac-preferences.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { systemPreferences, UserDefaultTypes } from 'electron';
+import { PlatformPreferences } from './preferences';
+
+/**
+ * MacOS specific implementation
+ *
+ * The preferences are stored in defauls, which is MacOS specific.
```
Comment (kevinushey) [inline src/node/desktop/src/main/preferences/mac-preferences.ts:7]:
```suggestion
 * The preferences are stored in defaults, which is MacOS specific.
```

File: src/node/desktop/src/main/preferences/preferences.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    case 'win32':
+      return new FilePreferences(`${homedir()}\\AppData\\Roaming\\RStudio\\desktop.ini`);
+    case 'linux':
+      return new FilePreferences(`${homedir()}/.config/RStudio/desktop.ini`);
+    default:
+      throw Error('unsupported platform');
+  }
+};
+
+export default getPreferenceManager;
```
Comment (kevinushey) [inline src/node/desktop/src/main/preferences/preferences.ts:37]: I think you could also just export an object called `preferenceManager` and initialize it when this module is loaded. IMHO that's marginally cleaner; do you have a strong opinion?

**Review by kevinushey — APPROVED** [review]
LGTM!

File: src/node/desktop/src/main/preferences/preferences.ts (hunk 2 above)
Comment (timtmok) [reply src/node/desktop/src/main/preferences/preferences.ts:37]: Yes, I think that would be a cleaner option! 👍

**Pull request merged**